src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -776,6 +769,12 @@ public function compile(/*$resolveEnvPlaceholders = false*/)
         }
 
         parent::compile();
+
+        foreach ($this->definitions + $this->aliasDefinitions as $id => $definition) {
+            if (!$definition->isPublic() || $definition->isPrivate()) {
+                $this->removedIds[$id] = true;
```

Comment:
This is for ppl doing that, which is a no-op, but one we catch now:
![image](https://user-images.githubusercontent.com/243674/33487334-252032ac-d6ad-11e7-839e-3fc51e17229e.png)